nucleus-sync/src/main/java/org/alfresco/hxi_connector/nucleus_sync/client/AlfrescoClient.java

```diff
@@ -0,0 +1,174 @@
+    {
+        return fetchAllPagedData(
+                "/people", new TypeReference<AlfrescoPagedResponse<AlfrescoUser>>() {}, "users");
+    }
+
+    public List<AlfrescoGroup> getAllGroups()
+    {
+        return fetchAllPagedData(
+                "/groups", new TypeReference<AlfrescoPagedResponse<AlfrescoGroup>>() {}, "groups");
+    }
+
+    public List<String> getUserGroups(String userId)
+    {
+        List<AlfrescoGroup> groups = fetchAllPagedData(
+                "/people/" + userId + "/groups",
+                new TypeReference<AlfrescoPagedResponse<AlfrescoGroup>>() {},
+                "groups for user " + userId);
+        return groups.stream().map(AlfrescoGroup::getId).toList();
+    }
+
+    private <T> List<T> fetchAllPagedData(
+            String basePath, TypeReference<AlfrescoPagedResponse<T>> typeRef, String errorContext)
+    {
+        try
+        {
+            List<T> results = new ArrayList<>();
+
+            int skipCount = 0;
+            boolean hasMoreItems = true;
+
+            while (hasMoreItems)
+            {
+                String response = makeAuthenticatedRequest(basePath + "?maxItems=100&skipCount=" + skipCount)
+                        .bodyToMono(String.class)
+                        .block(Duration.ofMinutes(timeoutInMins));
+
+                AlfrescoPagedResponse<T> pagedResponse = objectMapper.readValue(response, typeRef);
+
+                int itemsProcessed = 0;
+                if (pagedResponse.getList() != null
+                        && pagedResponse.getList().getEntries() != null)
+                {
+                    for (AlfrescoPagedResponse.EntryWrapper<T> entry : pagedResponse.getList().getEntries())
+                    {
+                        if (entry.getEntry() != null)
+                        {
+                            results.add(entry.getEntry());
+                            itemsProcessed++;
+                        }
+                    }
+                }
+
+                AlfrescoPagedResponse.Pagination pagination = pagedResponse.getList().getPagination();
+
+                if (pagination != null)
+                {
+                    hasMoreItems = pagination.isHasMoreItems();
+                    skipCount += itemsProcessed;
+                }
+                else
+                {
+                    hasMoreItems = false;
+                }
+            }
+
+            return results;
+        }
+        catch (Exception e)
+        {
+            LOGGER.error("Error fetching " + errorContext + ": {}", e.getMessage(), e);
+            throw new ClientException("Failed to fetch " + errorContext, e);
+        }
+    }
+
+    @Retryable(
+            retryFor = {
+                    RuntimeException.class,
+                    WebClientRequestException.class,
+                    WebClientResponseException.class
+            },
+            maxAttempts = 3,
+            backoff = @Backoff(delay = 2000, multiplier = 3, maxDelay = 10000))
```

Comment:
I'm pretty sure we already have backoff config in our application.yml file somewhere. Anyway - this can also be fixed as part of the future PR.